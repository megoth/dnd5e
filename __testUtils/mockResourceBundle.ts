import { mockSolidDatasetFrom } from "@inrupt/solid-client";
import { ResourceBundleModel } from "../src/models/resourceBundle";
import mockFluentBundle from "./mockFluentBundle";
import { createSWRResponse } from "./mockSWR";
import { appVocabURL } from "./mockAppIndexDataset";

export const errorsIndexURL = "https://example.com/errors.ttl";
export const faqIndexURL = "https://example.com/faq.ttl";
export const translationsIndexURL = "https://example.com/translations.ttl";
export const localizedIndexURL = "https://example.com/translations.en-US.ttl";

export default function mockResourceBundle(
  overrides: Partial<ResourceBundleModel> = {}
): ResourceBundleModel {
  return {
    appVocabURL,
    bundleNames: ["global"],
    currentLanguage: "en-US",
    errorsIndexURL: { global: errorsIndexURL },
    errorsIndexSWR: {
      global: createSWRResponse(mockSolidDatasetFrom(errorsIndexURL)),
    },
    faqIndexURL: { global: faqIndexURL },
    faqIndexSWR: {
      global: createSWRResponse(mockSolidDatasetFrom(faqIndexURL)),
    },
    fluentBundles: {
      global: [mockFluentBundle({}, "en-US")],
    },
    localizedIndexURL: { global: localizedIndexURL },
    localizedIndexSWR: {
      global: createSWRResponse(mockSolidDatasetFrom(faqIndexURL)),
    },
    translationsIndexURL: { global: translationsIndexURL },
    translationsIndexSWR: {
      global: createSWRResponse(mockSolidDatasetFrom(translationsIndexURL)),
    },
    ...overrides,
  };
}
