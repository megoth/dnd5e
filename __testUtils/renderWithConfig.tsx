import React from "react";
import { render } from "@testing-library/react";
import AppConfigProvider from "../src/contexts/appConfig";
import mockAppConfig from "./mockAppConfig";

export default function renderWithConfig(ui) {
  const { errorsUrl, translationsUrl } = mockAppConfig();
  return render(
    <AppConfigProvider errorsUrl={errorsUrl} translationsUrl={translationsUrl}>
      {ui}
    </AppConfigProvider>
  );
}
