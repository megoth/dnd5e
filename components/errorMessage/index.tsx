import React from "react";
import { Localized } from "@fluent/react";
import NestedError from "nested-error-stacks";
import { getErrorId } from "../../src/models/error";

interface Props {
  error: NestedError;
}

export const TESTID_ERROR = "error";
export const TESTID_ERROR_TITLE = "error-title";

export default function ErrorMessage({ error }: Props) {
  const { message: errorUrl } = error;
  const errorId = getErrorId(errorUrl);
  return (
    <div data-testid={TESTID_ERROR}>
      <h1 data-testid={TESTID_ERROR_TITLE}>
        <span resource={errorUrl}>
          <Localized id={errorId}>{errorUrl}</Localized>
        </span>
      </h1>
    </div>
  );
}
