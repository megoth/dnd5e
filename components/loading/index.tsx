import React from "react";

export const TESTID_LOADING = "loading";

export default function Loading() {
  // Do not try to translate anything here - it's used when loading translations as well
  return <div data-testid={TESTID_LOADING}>Loading...</div>;
}
