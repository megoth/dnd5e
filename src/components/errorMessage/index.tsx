import React from "react";
import NestedError from "nested-error-stacks";

interface Props {
  error: NestedError;
}

export const TESTID_ERROR = "error";
export const TESTID_ERROR_TITLE = "error-title";

export default function ErrorMessage({ error }: Props) {
  // const { message: errorURL } = error;
  return (
    <div data-testid={TESTID_ERROR}>
      <h1 data-testid={TESTID_ERROR_TITLE}>
        {/*{translationURL ? (*/}
        {/*  <Localized id={getTranslationId(translationURL)}>*/}
        {/*    {errorURL}*/}
        {/*  </Localized>*/}
        {/*) : (*/}
        {error.message}
        {/*)}*/}
      </h1>
    </div>
  );
}
