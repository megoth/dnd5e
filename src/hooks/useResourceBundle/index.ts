import { useContext } from "react";
import { ResourceBundleContext } from "../../contexts/resourceBundle";

export default function useResourceBundle() {
  return useContext(ResourceBundleContext);
}
