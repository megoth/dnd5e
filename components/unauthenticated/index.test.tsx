import React from "react";
import Unauthenticated, { setupLoginSubmit } from "./index";
import { TESTID_LOADING } from "../loading";
import { TESTID_ERROR } from "../errorMessage";
import renderWithConfig from "../../__testUtils/renderWithConfig";
import useResourceBundle from "../../src/hooks/useResourceBundle";
import mockResourceBundle from "../../__testUtils/mockResourceBundle";

jest.mock("../../src/hooks/useResourceBundle");
const mockedResourceBundleHook = useResourceBundle as jest.Mock;

describe("Unauthenticated", () => {
  it("renders a login form", async () => {
    mockedResourceBundleHook.mockReturnValue({ bundle: mockResourceBundle() });
    const { asFragment } = renderWithConfig(<Unauthenticated />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders loading when translations are not loaded yet", () => {
    mockedResourceBundleHook.mockReturnValue({});
    const { asFragment, getByTestId } = renderWithConfig(<Unauthenticated />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTID_LOADING)).toBeDefined();
  });

  it("renders an error message if something goes wrong", () => {
    mockedResourceBundleHook.mockReturnValue({
      error: new Error("error"),
    });
    const { asFragment, getByTestId } = renderWithConfig(<Unauthenticated />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTID_ERROR)).toBeDefined();
  });
});

describe("setupLoginSubmit", () => {
  it("handles login", () => {
    const login = jest.fn();
    const event = { preventDefault: jest.fn() };
    expect(setupLoginSubmit(login)(event)).toBeUndefined();
    expect(event.preventDefault).toHaveBeenCalledWith();
    expect(login).toHaveBeenCalledWith({});
  });
});
