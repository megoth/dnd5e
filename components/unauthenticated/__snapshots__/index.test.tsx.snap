// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unauthenticated renders 1`] = `
<DocumentFragment>
  <div
    class="text-center"
  >
    <h2
      class="font-bold text-xl"
    >
      [Translation for https://example.com/global/translations.ttl#recommendedIdPSet does not exist]
    </h2>
    <ul
      class="flex flex-col space-y-2 my-2"
    >
      <li>
        <button
          class="button button--solid w-full"
          data-testid="login-button"
          type="button"
        >
          Inrupt Pod Spaces
        </button>
      </li>
      <li>
        <button
          class="button button--solid w-full"
          data-testid="login-button"
          type="button"
        >
          solidcommunity.net
        </button>
      </li>
      <li>
        <button
          class="button button--solid w-full"
          data-testid="login-button"
          type="button"
        >
          inrupt.net
        </button>
      </li>
    </ul>
    <h2
      class="line-behind font-bold text-xl"
    >
      <span
        class="line-behind__line"
      />
      <span
        class="bg-white dark:bg-gray-800 lg:bg-white lg:dark:bg-gray-800"
      >
        [Translation for https://example.com/global/translations.ttl#or does not exist]
      </span>
    </h2>
    <form
      class="mt-2 md:mt-0 text-left"
    >
      <img
        alt="[Translation for https://example.com/global/translations.ttl#logInImage does not exist]"
        class="mx-auto hidden md:block"
        src="/logos/solid-emblem.svg"
        style="max-width: 200px;"
      />
      <label
        class="label"
        for="idp"
      >
        [Translation for https://example.com/global/translations.ttl#solidIdP does not exist]
        <input
          class="input"
          data-testid="login-form-idp-field"
          id="idp"
          placeholder="https://"
          required=""
          type="url"
          value=""
        />
      </label>
      <label
        class="label"
        for="rememberIdp"
      >
        <input
          class="checkbox"
          data-testid="login-form-remember-checkbox"
          id="rememberIdp"
          type="checkbox"
        />
        <span
          class="label__text"
        >
          [Translation for https://example.com/global/translations.ttl#rememberIdP does not exist]
        </span>
      </label>
      <button
        class="button w-full"
        data-testid="login-form-button"
        type="submit"
      >
        [Translation for https://example.com/global/translations.ttl#logIn does not exist]
      </button>
      <div
        class="mt-4"
      >
        <a
          class="link"
          href="/signup"
        >
          [Translation for https://example.com/global/translations.ttl#signupSolidPrompt does not exist]
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
